nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/ListBoxFileMetadataTemplates.java

```diff
@@ -0,0 +1,231 @@
+                flowFile = session.putAttribute(flowFile, "box.metadata.templates.count", "0");
+                session.transfer(flowFile, REL_SUCCESS);
+                return;
+            }
+
+            while (iterator.hasNext()) {
+                final Metadata metadata = iterator.next();
+                final Map<String, Object> templateFields = new HashMap<>();
+
+                templateNames.add(metadata.getTemplateName());
+
+                // Add standard metadata fields
+                templateFields.put("$id", metadata.getID());
+                templateFields.put("$type", metadata.getTypeName());
+                templateFields.put("$parent", "file_" + fileId); // match the Box API format
+                templateFields.put("$template", metadata.getTemplateName());
+                templateFields.put("$scope", metadata.getScope());
+
+                // Add all dynamic fields from the metadata
+                for (final String fieldName : metadata.getPropertyPaths()) {
+                    if (metadata.getValue(fieldName) != null) {
+                        String cleanFieldName = fieldName.startsWith("/") ? fieldName.substring(1) : fieldName;
+                        String fieldValue = metadata.getValue(fieldName).asString();
+                        templateFields.put(cleanFieldName, fieldValue);
+                    }
+                }
+
+                templatesList.add(templateFields);
+            }
+
+            try {
+                try (final OutputStream out = session.write(flowFile);
+                     final BoxMetadataJsonArrayWriter writer = BoxMetadataJsonArrayWriter.create(out)) {
+
+                    // Write each metadata template as a separate JSON object in the array
+                    for (Map<String, Object> templateFields : templatesList) {
+                        writer.write(templateFields);
+                    }
+                }
+
+                final Map<String, String> recordAttributes = new HashMap<>();
+                recordAttributes.put("record.count", String.valueOf(templatesList.size()));
+                recordAttributes.put(CoreAttributes.MIME_TYPE.key(), "application/json");
+                recordAttributes.put("box.file.id", fileId);
+                recordAttributes.put("box.metadata.templates.names", String.join(",", templateNames));
+                recordAttributes.put("box.metadata.templates.count", String.valueOf(templatesList.size()));
+                flowFile = session.putAllAttributes(flowFile, recordAttributes);
+
+                session.getProvenanceReporter().receive(flowFile, BoxFileUtils.BOX_URL + fileId);
+                session.transfer(flowFile, REL_SUCCESS);
+            } catch (final IOException e) {
+                getLogger().error("Failed writing metadata templates from file [{}]", fileId, e);
+                flowFile = session.putAttribute(flowFile, ERROR_MESSAGE, e.getMessage());
+                flowFile = session.penalize(flowFile);
```

Comment:
I don't think I would penalize the flowfile here, if the user wants to retry the flowfile with a penalty duration, the user should use the retry mechanism at framework level instead of using a self loop relationship.